File: src/main/java/RemoveElement/GoKu-gaga/RemoveElement.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
  for(let i = 0; i < nums.length; i++) {
      if(nums[i] === val) {
          nums.splice(i, 1);
          i -= 1;
      }
  }
  return nums.length;
};

/** 
 * 执行结果
 * 执行用时 : 76 ms, 在Remove Element的JavaScript提交中击败了96.95% 的用户
 * 内存消耗 : 33.6 MB, 在Remove Element的JavaScript提交中击败了54.51% 的用户
*/

File: src/main/java/RemoveElement/Zntao/RemoveElement.java
package RemoveElement.Zntao;

/**
 * Created by Zntao
 */
public class RemoveElement {
    //  与昨天题目思路类似，将0替换成val
    public static int removeElement(int[] nums, int val) {

        int length = 0;
        int seq = 0;
        for(int i = 0;i < nums.length; i++){
            if(nums[i] != val){
                int tem = nums[i];
                nums[i] = val;
                nums[seq] = tem;
                seq++;
                length++;
            }
        }

        return length;

    }

    public static void main(String[] args) {

        int a[] = {3,2,2,3};
        int length = removeElement(a, 3);
        System.out.println(length);
        /*
        * Runtime: 0 ms, faster than 100.00% of Java online submissions for Remove Element.
          Memory Usage: 34.3 MB, less than 100.00% of Java online submissions for Remove Element.
        * */
    }
}
